src/gmail.js

```diff
@@ -3843,23 +3864,45 @@ var Gmail = function(localJQuery) {
         */
         recipients: function(options) {
             if( typeof options !== "object" ) options = {};
-            var name_selector = options.type ? "[name=" + options.type + "]" : "";
-
-            // determine an array of all emails specified for To, CC and BCC and extract addresses into an object for the callback
-            var recipients = options.flat ? [] : { to: [], cc: [], bcc: [] };
-            this.$el.find(".GS input[type=hidden]"+name_selector).each(function(idx, recipient ){
-                if(options.flat) {
-                    recipients.push(recipient.value);
+            const peopleKit = api.check.is_peoplekit_compose(this.$el);
+            
+            const type_selector = options.type ?
+                peopleKit ?
+                    "div[name=" + options.type + "] " :
+                    "[name=" + options.type + "]" : "";
+
+            const found = peopleKit ?
+                this.$el.find("tr.bzf " + type_selector + "div[data-hovercard-id]").map((_, el) => ({
+                    type: el.closest("div[name]").getAttribute("name"),
+                    email: el.getAttribute("data-hovercard-id")
+                })) :
+                this.$el.find(".GS input[type=hidden]" + type_selector).map((_, el) => ({
+                    type: el.name,
+                    email: el.value
+                }));
+
+            if (options.flat) {
+                return found.toArray().map(r => r.email);
+            } else {
+                let result = { to: [], cc: [], bcc: [] };
+                if (options.type) {
+                    result[options.type] = found.toArray()
+                        .filter(r => r.type === options.type)
+                        .map(r => r.email);
                 } else {
-                    if(!recipients[recipient.name]) recipients[recipient.name] = [];
-                    recipients[recipient.name].push(recipient.value);
+                    ["to", "cc", "bcc"].forEach(type => {
+                        result[type] = found.toArray()
+                            .filter(r => r.type === type)
+                            .map(r => r.email);
+                    });
                 }
-            });
-            return recipients;
+                return result;
+            }
         },
 
         /**
-           Retrieve the current "to" recipients
+           Retrieve the typing area for "to" recipients, not recipients. 
+           Either textarea or input, which can be empty if last recipient are typed and selected (by pressing ENTER)
```

Comment:
The documentation is very clear on "retrieve", but the API code in the past has been bi-directional.

That is, these calls could get *and* set the value in the approiate compose-field. Is this still true with this PeopleKit implementation?

Reply:
It is not even true with pre-peoplekit - textarea this thing refers to gets cleared immediately once you enter something and press ENTER

So you can obtain only intermediate value, but you can set value

Reply:
Oh well. In that case it still sounds like an improvement, and I'm not going to block this PR over this :smile: